deployer/scripts/hawkular.sh

```diff
@@ -187,12 +187,25 @@ EOF
   oc create -f templates/hawkular-cassandra-node-pv.yaml
   oc create -f templates/hawkular-cassandra-node-emptydir.yaml
   oc create -f templates/support.yaml
-  
+
   echo "Deploying Hawkular Metrics & Cassandra Components"
   oc process hawkular-metrics -v "IMAGE_PREFIX=$image_prefix,IMAGE_VERSION=$image_version,METRIC_DURATION=$metric_duration,MASTER_URL=$master_url" | oc create -f -
   oc process hawkular-cassandra-services | oc create -f -
-  oc process hawkular-support -v "HAWKULAR_METRICS_HOSTNAME=$hawkular_metrics_hostname" | oc create -f - || true
-  
+  oc process hawkular-support | oc create -f -
+
+  echo "Creating the Hawkular Metrics Route"
+  # We need to create the route after the service has been created so that its labels get applied to the route itself
+  route_params="--hostname=$hawkular_metrics_hostname --service=hawkular-metrics --dest-ca-cert=$dir/hawkular-metrics-ca.cert"
+  if [ -s ${secret_dir}/hawkular-metrics.pem ]; then
+    `openssl rsa -in  ${secret_dir}/hawkular-metrics.pem > $dir/custom-certificate.key`
+    `openssl crl2pkcs7 -nocrl -certfile ${secret_dir}/hawkular-metrics.pem | openssl pkcs7 -print_certs | grep -v "^subject=*\|^issuer=*\|^$" > $dir/custom-certificate.crt`
```

Comment:
That command does not exactly do what we want it to do. The openssl command assumes a single certificate is in the pem file, which is not always true. The command above will extract all the certificates in the chain as part of the pem file.

When oadm generates certificates, they include both the CA certificate and the server one. This means we can just `cat myServer.crt myServer.key > myServer.pem` and we have all the information we need.

If the user get their certs from something else and their pem only contains the server key and the server certificate itself, they can still add in the certificate chain using the  hawkular-metrics-ca.cert secret.

I will add a comment explaining what we are doing here.